src/Acuminator/Acuminator.Analyzers/StaticAnalysis/DacKeyFieldDeclaration/KeyFieldDeclarationFix.cs

```diff
@@ -0,0 +1,142 @@
+																												diagnostic,
+																												CodeFixModes.RemoveIdentityAttribute),
+																			equivalenceKey: codeActionRemoveIdentityColumnName);
+
+			context.RegisterCodeFix(codeActionIdentityKey, context.Diagnostics);
+			context.RegisterCodeFix(codeActionBoundKeys, context.Diagnostics);
+			context.RegisterCodeFix(codeActionRemoveIdentityColumn, context.Diagnostics);
+
+			return Task.FromResult(true);
+		}
+
+		private async Task<Document> RemoveKeysFromFieldsAsync(Document document,
+																CancellationToken cancellationToken,
+																Diagnostic diagnostic,
+																CodeFixModes mode)
+		{
+
+			SemanticModel semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+			cancellationToken.ThrowIfCancellationRequested();
+
+			Location[] attributeLocations = diagnostic.AdditionalLocations.ToArray();
+
+			SyntaxNode root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+			List<SyntaxNode> deletedNodes = new List<SyntaxNode>();
+
+			foreach (var attributeLocation in attributeLocations)
+			{
+
+				if (!(root?.FindNode(attributeLocation.SourceSpan) is AttributeSyntax attributeNode))
+					return document;
+
+				ITypeSymbol attributeType = semanticModel.GetTypeInfo(attributeNode, cancellationToken).Type;
+
+				if (attributeType == null)
+					return document;
+
+				var attributeInformation = new AttributeInformation(new PXContext(semanticModel.Compilation));
+				bool isIdentityAttribute = attributeInformation.IsAttributeDerivedFromClass(attributeType, new PXContext(semanticModel.Compilation).FieldAttributes.PXDBIdentityAttribute) ||
+										   attributeInformation.IsAttributeDerivedFromClass(attributeType, new PXContext(semanticModel.Compilation).FieldAttributes.PXDBLongIdentityAttribute);
+
+
+				if ((mode == CodeFixModes.EditIdentityAttribute && isIdentityAttribute) ||
+					(mode == CodeFixModes.EditKeyFieldAttributes && !isIdentityAttribute))
+				{
+					IEnumerable<AttributeArgumentSyntax> deletedNode = GetIsKeyEQTrueArguments(attributeNode);
+
+					deletedNodes.AddRange(deletedNode);
+				}
+				if (mode == CodeFixModes.RemoveIdentityAttribute && isIdentityAttribute)
+				{
+					if ((attributeNode.Parent as AttributeListSyntax).Attributes.Count == 1)
+						deletedNodes.Add(attributeNode.Parent);
+					else
+						deletedNodes.Add(attributeNode);
+				}
+			}
+
+			SyntaxNode newRoot;
+			if (mode == CodeFixModes.RemoveIdentityAttribute)
+				newRoot = root.RemoveNodes(deletedNodes, SyntaxRemoveOptions.KeepExteriorTrivia);
+			else
+				newRoot = root.RemoveNodes(deletedNodes, SyntaxRemoveOptions.KeepNoTrivia);
+
+			return document.WithSyntaxRoot(newRoot);
+		}
+
+		private IEnumerable<AttributeArgumentSyntax> GetIsKeyEQTrueArguments(AttributeSyntax attributeNode)
+		{
+			return attributeNode.ArgumentList.Arguments.Where(a => a.NameEquals?.Name.Identifier.ValueText.Equals(IsKey) ?? false &&
```

Comment:
@saloponov the predicate is hard to read and operators precedence order is not so obvious. Consider:
```
IsKey == a.NameEquals?.Name.Identifier.ValueText && bool.TrueString == (a.Expression as LiteralExpressionSyntax)?.Token.ValueText    
```

Reply:
@SENya1990 Your decision isn't working.